Sources/SourceKitLSP/Swift/SwiftLanguageService.swift

```diff
@@ -1091,6 +1111,18 @@ extension DocumentSnapshot {
     return lowerBound..<upperBound
   }
 
+  /// Extracts the range of the given syntax node in terms of positions within
+  /// this source file.
+  func range(
+    of node: some SyntaxProtocol,
+    callerFile: StaticString = #fileID,
+    callerLine: UInt = #line
+  ) -> Range<Position> {
+    let lowerBound = self.position(of: node.position)
+    let upperBound = self.position(of: node.endPosition)
```

Comment:
I think you should pass `callerFile` and `callerLine` into `position(of:)` so that the logger fault message contains the file and line number that is the real originator if position conversion fails.